HWDataset.py

```diff
@@ -0,0 +1,90 @@
+from torch.utils.data import Dataset
+from torch.utils.data import DataLoader
+from pathlib import Path
+import math
+import numpy as np
+import plot_stroke
+import torch
+
+def collate_fn(samples):
+    batch_size=len(samples)
+    strokes = [s for s, text in samples]
+    texts = [text for s, text in samples]
+
+    max_stroke_len = max([len(s) for s in strokes])
+    num_stroke_elements=strokes[0].shape[1]
+    all_strokes = torch.zeros((max_stroke_len, batch_size, num_stroke_elements))
+    strokes_mask = torch.ones((max_stroke_len, batch_size), dtype=torch.bool)
+    for i in range(batch_size):
+        this_stroke_len = strokes[i].shape[0]
+        # pad
+        all_strokes[:this_stroke_len,i] = torch.tensor(strokes[i])
```

Comment:
Do you know about this?
`from torch.nn.utils.rnn import pad_sequence`
It can be useful to refactor this code a bit.

Reply:
Very nice.